src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -220,7 +219,8 @@ public override async Task<DValue<BooleanValue>> RemoveAsync(IEnumerable<Formula
 
                 if (!found)
                 {
-                    errors.Add(CommonErrors.RecordNotFound());
+                    // !!!TODO This validation should be moved to the interpreter.

```

Comment:
>!!!TODO This validation should be moved to the interpreter. [](http://example.com/codeflow?start=23&length=59)

Should we move the entire folder (Values) to the interpreter? The values are a runtime concept, it should not be needed during compile-time.

Reply:
I ran a quick test to see how big would be the damage moving the value to interpreter. Not as big as I expected. This might be doable but in a separate PR.

Reply:
definitely separate PR. 

It would also be a massive breaking change.  (moving classes between assemblies is breaking).